### PR TITLE
feat: add lit-actions-ts-bundling example

### PR DESCRIPTION
**Pull request by Ansonhkg**
# What

An example to write Lit Action code in Typescript and bundling them using esBuild. You will also get Type definition for the Lit Action global variables, such as `Lit.Actions` and `Lit.Auth`. 

image

---

# Getting started

```
npm install
```

## Start writing your Lit Action like a regular TS

- See demo in `./src/lit-actions/hello-world.ts`

## To build and bundle the TS Lit Action code

```
npm run build
```

This will build and bundle all files in `./src/lit-actions/*.ts`, and the output will be in `./src/lit-actions/dist`.

## Getting the bundled Lit Action Code

In your `./src/index.ts` file, you can import the bundled Lit Action code like this:

```ts
import { code } from "./lit-actions/dist/hello-world.js";

console.log("litActionCode:", code);
```

## output

```
npm run build

> lit-actions-ts-bundling@1.0.0 build
> node esbuild.config.js

🗂️  File: src/lit-actions/dist/hello-world.js
   Size: 0.0251 MB (decimal) | 0.0239 MB (binary)
================================================
✅ Lit actions built successfully in 0.02 seconds
```

### CODE REVIEW

File: templates/browser/src/App.tsx
Comment (spacesailor24) [inline templates/browser/src/App.tsx:1]: Was thing for testing, or was the intention to update the template?

File: session-signatures/browser/.vercel/README.txt
Comment (spacesailor24) [inline session-signatures/browser/.vercel/README.txt:1]: Was this and `project.json` intended to be committed?

File: templates/browser/src/App.tsx
Comment (Ansonhkg) [reply templates/browser/src/App.tsx:1]: Oh yea we need to revert to this, it's my local testing for my own pkg lol

File: session-signatures/browser/.vercel/README.txt
Comment (Ansonhkg) [reply session-signatures/browser/.vercel/README.txt:1]: Same as above

File: lit-actions-ts-bundling/esbuild.config.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+          .replace(/var import_ethers = __require\("ethers"\);/g, "")
+          .replace(/import_ethers\./g, "");
+
+        // IMPORTANT: if minify is enabled, we need to:
+        // 1. remove var t=o(\"ethers\");
+        // 2. replace t.ethers to ethers
+        // TODO: This is not working - must fix soon.
+        content = content
+          .replace(/var [a-zA-Z]=o\("ethers"\);/g, "")
+          .replace(/[a-zA-Z]\.ethers/g, "ethers");
```
Comment (Ansonhkg) [inline lit-actions-ts-bundling/esbuild.config.js:75]:
```suggestion
        content = content
          .replace(/var\s+\w+=\w+\("ethers"\);/g, '')
          .replace(/[a-zA-Z]\.ethers/g, "ethers");
```

File: lit-actions-ts-bundling/esbuild.config.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      });
+    });
+  },
+};
+
+const promises = configs.map((config) => {
+  return esbuild.build({
+    entryPoints: [config.entryPoint],
+    bundle: true,
+    minify: false, // TODO: Only turn this on fix replacing is working
```
Comment (Ansonhkg) [inline lit-actions-ts-bundling/esbuild.config.js:102]:
```suggestion
    minify: false, // Up to user to turn it on/off. Default off.
```

**Comment by Ansonhkg** [comment]
don't merge this yet

**Review by spacesailor24 — APPROVED** [review]

**Pull request merged**